### ISSUE TITLE
Error with plugin on release version of APK

### ISSUE THREAD

**Issue by pedromareis** [NativeScript/NativeScript#4861]
Hey guys, I' having some trouble using a .jar file on my NS project.

Basically I've created a plugin "nativescript-bt-printer" which only has the package.json, the index.js (empty file) and the jar file on the platforms/android/libs folder. When I add the plugin using _**tns plugin add C:/Users/.../nativescript-bt-printer**_ it all goes as supposed and the app runs perfectly BUT when I build the app to publish on the AppStore, it crashes when it tries to use the functions of the .jar...

Can anyone help me?

Best regards,
Pedro Reis

**Comment by tsonevn** [comment]
Hi @pedromareis,
First of all, thank you for your interest in NativeScirpt.

Regarding the topic, it would help if you could provide sample project and together with the plugin, which will allow us to recreate the problem and to debug it locally

**Comment by pedromareis** [comment]
Hello @tsonevn, I've attached my plugin which contains the jar library to this post. I've also tried to publish it and use it as a NPM package but I can't even instantiate the class of the jar.

Since my project is really huge, I'll post the part of code related to my problem:

```
var app = require("application");

var context = app.android.context;
var mHandler = new android.os.Handler;
var mService = new com.zj.btsdk.BluetoothService(context, mHandler);
```
The release apk crashes when creating the BluetoothService object.

**PLUGIN:**
[nativescript-bt-printer.zip](https://github.com/NativeScript/NativeScript/files/1320696/nativescript-bt-printer.zip)

**Comment by pedromareis** [comment]
I've finally solved this issue! Basically I've created a new Android Library, called the package org.nativescript.btprinter and put all the classes the .jar file had on the main folder. I've also imported the jar (adding the file to the libs folder AND importing it on the project structure). Then I've generated the .aar file.

After this, I've added the file to my custom nativescript-bt-printer plugin and added it using tns plugin add <path-to-plugin-folder> and created my BluetoothService object using var mService = new org.nativescript.btprinter.BluetoothService(context, mHandler);.
